src/OceanSeaIceModels/CrossRealmFluxes/atmosphere_ocean_fluxes.jl

```diff
@@ -17,46 +23,98 @@ function compute_atmosphere_ocean_fluxes!(coupled_model)
 
+    # kernel parameters that compute fluxes in 0:Nx+1 and 0:Ny+1
+    kernel_size = (size(grid, 1) + 2, size(grid, 2) + 2)
+    kernel_parameters = KernelParameters(kernel_size, (-1, -1))
+
+    surface_atmosphere_state = coupled_model.fluxes.surface_atmosphere_state
+
+    # Simplify NamedTuple to reduce parameter space consumption.
+    # See https://github.com/CliMA/ClimaOcean.jl/issues/116.
+    surface_atmosphere_state = (u = surface_atmosphere_state.u.data,
+                                v = surface_atmosphere_state.v.data,
+                                T = surface_atmosphere_state.T.data,
+                                p = surface_atmosphere_state.p.data,
+                                q = surface_atmosphere_state.q.data,
+                                Qs = surface_atmosphere_state.Qs.data,
+                                Qℓ = surface_atmosphere_state.Qℓ.data,
+                                Mp = surface_atmosphere_state.Mp.data)
+
+    launch!(arch, grid, kernel_parameters,
+            _interpolate_primary_atmospheric_state!,
+            surface_atmosphere_state,
+            grid,
+            clock,
+            atmosphere_velocities,
+            atmosphere_tracers,
+            atmosphere_pressure,
+            downwelling_radiation,
+            freshwater_flux,
+            atmosphere_grid,
+            atmosphere_times,
+            atmosphere_backend,
+            atmosphere_time_indexing)
+
+    # Separately interpolate the auxiliary freshwater fluxes, which may
+    # live on a different grid than the primary fluxes and atmospheric state.
     runoff_args = get_runoff_args(atmosphere.runoff_flux)
+    interpolated_prescribed_freshwater_flux = surface_atmosphere_state.Mp
+
+    if !isnothing(runoff_args)
+        launch!(arch, grid, kernel_parameters,
+                _interpolate_auxiliary_freshwater_fluxes!,
+                interpolated_prescribed_freshwater_flux,
+                grid,
+                clock,
+                runoff_args)
+    end
 
-    Qs = atmosphere.downwelling_radiation.shortwave
-    Ql = atmosphere.downwelling_radiation.longwave
+    #####
+    ##### Next compute turbulent fluxes.
+    #####
 
-    downwelling_radiation = (shortwave=Qs.data, longwave=Ql.data)
+     # Fluxes, and flux contributors
+    centered_velocity_fluxes = (u = coupled_model.fluxes.total.ocean.momentum.uᶜᶜᶜ,
+                                v = coupled_model.fluxes.total.ocean.momentum.vᶜᶜᶜ)
 
-    kernel_size = (size(grid, 1) + 2, size(grid, 2) + 2)
+    staggered_velocity_fluxes = (u = coupled_model.fluxes.total.ocean.momentum.u,
+                                 v = coupled_model.fluxes.total.ocean.momentum.v)
 
-    # kernel parameters that compute fluxes in 0:Nx+1 and 0:Ny+1
-    kernel_parameters = KernelParameters(kernel_size, (-1, -1))
+    net_tracer_fluxes    = coupled_model.fluxes.total.ocean.tracers
+    similarity_theory    = coupled_model.fluxes.turbulent
```

Comment:
Is this a typo?

Reply:
hmm

Reply:
no, this is just moving around a line that was already there

Reply:
`OceanSeaIceFluxes.turbulent` seems to be an abstraction representing the turbulent flux computation, which here is `SimilarityTheoryTurbulentFluxes` or something. What do you suggest as a better name for this property?

Reply:
I am not sure. I was a bit confused because turbulent does not necessarily tell me that that is the similarity theory struct. `turbulent` can be a lot of things, it is not very specific

Reply:
I just realized that this line change is not connected to this PR (just moved around) so probably we can discuss this elsewhere